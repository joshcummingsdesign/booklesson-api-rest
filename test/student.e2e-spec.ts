import { INestApplication } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import * as request from 'supertest';
import { AppModule } from '../src/app.module';

describe('Student (e2e)', () => {
  let app: INestApplication;

  beforeAll(async () => {
    const module: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = module.createNestApplication();

    app.init();
  });

  it('should create account', async () => {
    expect(1).toEqual(1);
  });

  // --- Student flow
  // -- Profile
  // Create account
  // Login
  // View profile
  // Update profile
  // -- Availability
  // View all teachers
  // View teacher availability
  // -- Lessons
  // Create a lesson
  // View all lessons
  // Cancel a lesson

  afterAll(async () => {
    await app.close();
  });
});
