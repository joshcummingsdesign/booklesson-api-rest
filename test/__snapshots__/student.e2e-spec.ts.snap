// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student (e2e) should create, read, delete lesson 1`] = `
Object {
  "datetime": "2020-06-20T10:00:00-07:00",
  "id": 1,
  "studentId": 998,
  "teacherId": 999,
}
`;

exports[`Student (e2e) should create, read, delete lesson 2`] = `
Array [
  Object {
    "datetime": "2020-06-20T09:00:00-07:00",
    "id": 999,
    "studentId": 998,
    "teacherId": 999,
  },
  Object {
    "datetime": "2020-06-20T10:00:00-07:00",
    "id": 1,
    "studentId": 998,
    "teacherId": 999,
  },
]
`;

exports[`Student (e2e) should create, read, delete lesson 3`] = `
Object {
  "datetime": "2020-06-20T10:00:00-07:00",
  "id": 1,
  "studentId": 998,
  "teacherId": 999,
}
`;

exports[`Student (e2e) should view all teachers 1`] = `
Array [
  Object {
    "email": "kurtrosenwinkel@gmail.com",
    "firstName": "Kurt",
    "id": 999,
    "instrument": "guitar",
    "lastName": "Rosenwinkel",
    "role": "teacher",
  },
]
`;

exports[`Student (e2e) should view teacher availability 1`] = `
Array [
  Object {
    "available": false,
    "datetime": "2020-06-20T09:00:00-07:00",
    "id": 998,
    "userId": 999,
  },
  Object {
    "available": true,
    "datetime": "2020-06-20T10:00:00-07:00",
    "id": 999,
    "userId": 999,
  },
]
`;
